content/en/docs/plugins/developers-guide.md

```diff
@@ -189,13 +183,25 @@ type Closer interface {
 	Close()
 }
 
+type InitSchema interface {
+	InitSchema() *SchemaInfo
+}
+
+type OpenParams interface {
+	OpenParams() ([]OpenParam, error)
+}
+
 type Progresser interface {
 	Progress(pState sdk.PluginState) (float64, string)
 }
 ```
 
 Source and extrator plugins can optionally implement the `sdk.Destroyer` interface. In that case, `Destroy()` will be called when the plugin gets destroyed and can be used to release any allocated resource.
 
+Source and extrator plugins can also optionally implement the `sdk.InitSchema` interface. In that case, `InitSchema()` will be used to to return a schema describing the data expected to be passed as a configuration during the plugin initialization. This follows the semantics documented for [`get_init_schema`](../plugin-api-reference/#const-char-plugin-get-init-schema-ss-plugin-schema-type-schema-type-required-no). Currently, the schema must follow the [JSON Schema specific](https://json-schema.org/), which in Go can also be easily auto-generated with external packages (e.g. [alecthomas/jsonschema](https:/github.com/alecthomas/jsonschema)).
```

Comment:
```suggestion
Source and extrator plugins can also optionally implement the `sdk.InitSchema` interface. In that case, `InitSchema()` will be used to to return a schema describing the data expected to be passed as a configuration during the plugin initialization. This follows the semantics documented for [`get_init_schema`](../plugin-api-reference/#const-char-plugin-get-init-schema-ss-plugin-schema-type-schema-type-required-no). Currently, the schema must follow the [JSON Schema specific](https://json-schema.org/), which in Go can also be easily auto-generated with external packages (e.g. [alecthomas/jsonschema](https:/github.com/alecthomas/jsonschema)).
```

Does the framework call `InitSchema()`  always?

Reply:
If implemented, yes. The way the Go SDK does this is by always implementing a default `plugin_get_init_schema` that returns no schema (and thus is ignored by the framework), which eventually calls `InitSchema()` if the plugin struct implements it.